core/src/test/scala/unit/kafka/admin/DeleteTopicTest.scala

```diff
@@ -307,4 +307,27 @@ class DeleteTopicTest extends ZooKeeperTestHarness {
       (key, count)
     }
   }
+
+  @Test
+  def testDeleteWhenDeleteTopicIsDisabled() {
+    val topicAndPartition = TopicAndPartition("test", 0)
+    val topic = topicAndPartition.topic
+
+    // creating a cluster with "delete.topic.enable" = "false"
+    val servers = createTestTopicAndCluster(topic, "false")
+
+    // mark the topic for deletion
+    AdminUtils.deleteTopic(zkUtils, "test")
+    TestUtils.waitUntilTrue(() => !zkUtils.pathExists(getDeleteTopicPath(topic)),
+      "Admin path /admin/delete_topic/%s path not deleted even if deleteTopic is disabled".format(topic))
+    // verify that topic test is untouched
+    TestUtils.waitUntilTrue(() => servers.forall(_.getLogManager().getLog(topicAndPartition).isDefined),
```

Comment:
- You shouldn't need to `waitUntilTrue`
- The message should be something like "Replicas are missing" (since they were created but then incorrectly removed on test failure)